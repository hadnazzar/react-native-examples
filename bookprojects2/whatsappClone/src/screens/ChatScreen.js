import React, { Component } from 'react';
import {
  Text,
  View,
  Button,
  FlatList,
  ImageBackground,
  KeyboardAvoidingView
} from 'react-native';
// import { getMockData } from '../services/api';
import Compose from '../components/Compose';
import Message from '../components/Message';
import { getMessages , postMessage } from '../services/api';




export default class ChatScreen extends Component {

  state = {
    messages: []
  }
  keyboardVerticalOffset = 80;


  componentWillMount() {
    // getMockData().then((messages) =>
    //   this.setState({
    //     messages
    //   })
    // )

    this.unsubscribeGetMessages = getMessages((snapshot) => {
      this.setState({ messages: Object.values(snapshot.val()) })
    })
  }

  componentWillUnmount() {
    this.unsubscribeGetMessages();
  }

  static navigationOptions = ({ navigation }) => ({
    title: `Chat with ${navigation.state.params.name}`
  })

  render() {
    return (
      // <ImageBackground
      //   style={styles.container}
      //   source={require('assets/imgs/background.png')}>
      //   <FlatList
      //     data={this.state.messages}
      //     renderItem={({ item }) =>
      //       <View>
      //         <Text> {item.message} </Text>
      //       </View>
      //     }
      //   />
      // </ImageBackground>


      // <ImageBackground
      //   style={styles.container}
      //   source={require('../assets/imgs/background.png')}>
      //   <FlatList
      //     data={this.state.messages}
      //     renderItem={({ item }) =>
      //       this.getMessageRow(item)
      //     }
      //     keyExtractor={(item, index) => (`message-${index}`)}
      //   />
      // </ImageBackground>



      <ImageBackground
        style={styles.container}
        source={require('../assets/imgs/background.png')}>
        <KeyboardAvoidingView behavior="padding"
          keyboardVerticalOffset={this.keyboardVerticalOffset}
          style={styles.container}>

          <FlatList
            data={this.state.messages}
            renderItem={({item}) => <Message item={item}/>
            }
            keyExtractor={(item, index) => (`message-${index}`)}
          />
          <Compose submit={postMessage} />

        </KeyboardAvoidingView>
      </ImageBackground>
    )
  }
}


const styles = {
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    // justifyContent: 'center',
    // alignItems: 'center'
  }
}