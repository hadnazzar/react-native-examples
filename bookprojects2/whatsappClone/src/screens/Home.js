import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button
} from 'react-native';

export default class Home extends Component {

  static navigationOptions = {
    title: "Home Screen"
  }

  render() {
    return (
      <View style={styles.container}>
        <Button
          onPress={() => this.props.navigation.navigate('chat', { name: 'John' })}
          title="press"
        />
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center'
  }
}