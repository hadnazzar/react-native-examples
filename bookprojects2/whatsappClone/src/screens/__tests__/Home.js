import 'react-native';
import React from 'react';
import routes from '../../config/routes';
import Home from '../Home';
import ChatScreen from '../ChatScreen';

it('renders correctly', () => {
  const tree = renderer.create(
    <Home />
  );
  expect(tree).toMatchSnapshot();
})

it('has proper navigation options', () => {
  expect(Home.navigationOptions).toMatchSnapshot();
})