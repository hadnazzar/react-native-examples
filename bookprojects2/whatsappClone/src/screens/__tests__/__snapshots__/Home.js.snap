// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `
Object {
  "title": "Home Screen",
}
`;
